### PR TITLE
Instrument Repo for Code Coverage

### PR DESCRIPTION
**Pull request by jshcrowthe**
This is going to probably take a little extra time (we have to get the extra CI integration approved) but having this instrumented will give us a path to improving our coverage.

### Browser

All of our browser builds running through `karma-webpack` are instrumented via [`istanbul-instrumenter-loader`](https://www.npmjs.com/package/istanbul-instrumenter-loader). We then dump this info for each component into the component `coverage/browser` dir by browser.

### Node

All of our node builds are instrumented via `nyc` which is a tool provided by the istanbul folks. Unfortunately, karma is not supported in the same way so we needed two tools. Coverage for this tool is dumped in the `coverage/node` directory.

## Merging `lcov.info` Files

After collecting `lcov.info` files (i.e. the coverage reports) for each file we leverage [`lcov-result-merger`](https://github.com/mweibel/lcov-result-merger) to consolidate to a single file that we can then pipe to CI.

I am going to leave the CI integration (i.e. `coveralls`) in as @vikrum is currently evaluating if we can get access here.

### CODE REVIEW

**Comment by schmidt-sebastian** [comment]
Approved for RTDB. Got the coverage numbers that I needed. Thanks!

File: config/webpack.test.js [hunk 1]
```diff
@@ -27,9 +27,18 @@ module.exports = {
         use: 'ts-loader'
       },
       {
-        test: /\.js$/,
-        use: ['source-map-loader'],
+        test: /\.[tj]sx?$/,
```
Comment (wilhuff) [inline config/webpack.test.js]: Is this just a standard configuration now or are we really doing jsx and tsx stuff? In contrast in other locations we're still using just straight `.ts` file extensions.

Comment (jshcrowthe) [reply config/webpack.test.js]: It's just a standard thing, we have no need for jsx/tsx so it isn't that big of a deal I don't think.

**Comment by mikelehen** [comment]
FYI- It seems like you upgraded prettier in this PR, which caused a bunch of extra formatting changes, which is causing me merge conflicts in a different branch.  Going forward it would be helpful if:

1. We don't upgrade prettier except in an isolated PR that only contains the upgrade (and the fallout formatting changes)
2. We coordinate this across the team so that folks are aware (and potentially we get large PRs checked in before the upgrade to avoid conflicts).

**Pull request merged**